### ISSUE TITLE
GitHub connector: naming conflict with common variable names

### ISSUE THREAD

**Issue by MaxTru** [camunda/connectors#1379]
__Describe the Bug__

The GitHub Connector internally works with non-namespaced variables. Hence clashes can easily happen, which will create errors when trying to run the process.

__Steps to Reproduce__

1. Have a process variable called `issueNumber`
2. Use the GitHub Post Issue Comment Connector
3. Now you will construct the input something like this:
```
          <zeebe:input source="=string(issueNumber)" target="issueNumber" />
          <zeebe:input source="=baseUrl + &#34;/repos/&#34; + owner + &#34;/&#34; + repo + &#34;/issues/&#34; + issueNumber + &#34;comments&#34;" target="url" />
```
4. Now the URL cannot be constructed:
<img width="1867" alt="Bildschirmfoto 2023-11-07 um 11 00 11" src="https://github.com/camunda/connectors/assets/42800119/fb312b4a-6c17-49f4-9bdc-af9e4beb46f8">

=> I think if working with namespaced internal variables, the problem might be solved. E.g., use `connectors_issueNumber`

__Expected Behavior__

I can use whatever process variable name, and the Connectors still work

__Environment__

SaaS as of 07.11.

**Comment by sbuettner** [comment]
Yes, this is a known problem and has been experienced before in https://github.com/camunda/connectors/issues/1124

**Comment by sbuettner** [comment]
Here is a related zeebe issue that could solve this issue: https://github.com/camunda/zeebe/issues/11163

**Comment by MaxTru** [comment]
> Here is a related zeebe issue that could solve this issue: [camunda/zeebe#11163](https://github.com/camunda/zeebe/issues/11163)

This would be super safe.

However, until we have camunda/zeebe#11163, Connectors team could simply prefix their internal variables, isn't it? (e.g., use `connectorsInternal_issueNumber` instead of `issueNumber`). In theory this could still break, but the likelihood would be super low.

**Comment by sbuettner** [comment]
Yes, it would reduce the probability but we would have to rewrite all connectors and request models and use underscores in our variables/json mappings.

**Comment by sbuettner** [comment]
As discussed with the team in the grooming session, we would like to see proper local variable isolation support than introduce prefixes as this might either break existing element templates or wouldnt improvide the situation as the backwards compatibility would suffer from the same issues (input name like "body" could still be used in a different process variable)
